src/Core/Checkout/Cart/CartCompressor.php

```diff
@@ -72,4 +79,11 @@ public function unserialize(string $value, int $compressionMethod): mixed
 
         return unserialize($uncompressed);
     }
+
+    private function checkSerializationMaxSize(string $compressed): void
+    {
+        if ($this->serializationMaxSize && \strlen($compressed) > $this->serializationMaxSize * 1024 * 1024) {
```

Comment:
\strlen is not accurate
```suggestion
        if ($this->serializationMaxSize && \mb_strlen($compressed, '8bit') > $this->serializationMaxSize * 1024 * 1024) {
```